core/interfaces/src/consensus.rs

```diff
@@ -39,8 +41,18 @@ pub trait ConsensusInterface<C: Collection>:
         app: ::ApplicationInterface,
         broadcast: ::BroadcastInterface,
         archive: ::ArchiveInterface,
+        notifier: ::NotifierInterface,
     ) {
-        let executor = app.transaction_executor();
+        let executor = app
+            .transaction_executor()
+            .expect("ConsensusInferface::_init - Update Socket should be available");
+
+        let epoch_change_notifier = notifier.epoch_emitter();
+        executor.inject(move |res| {
+            if res.change_epoch {
+                epoch_change_notifier.epoch_changed();
+            }
+        });
```

Comment:
Part of the reasoning we want to move to this new notifier flow is because this injection we do into the Affair socket is overhead that we really dont want. So we want to pass this notifier to consensus and be notifying in `submit_batch` when we know the epoch has changed. Ill sync with you and give you some more color on what im describing off of github.

Reply:
yea, let's get rid of this flow here and instead trigger the event from the consensus.